test/new-e2e/tests/agent-platform/common/agent_behaviour.go

```diff
@@ -314,22 +314,36 @@ func CheckCWSBehaviour(t *testing.T, client *TestClient) {
 	t.Run("system-probe and security-agent communicate", func(tt *testing.T) {
 		var statusOutputJSON map[string]any
 		var result bool
-		for try := 0; try < 10 && !result; try++ {
+		for try := 1; try <= 20 && !result; try++ {
 			status, err := client.Host.Execute("sudo /opt/datadog-agent/embedded/bin/security-agent status -j")
-			if err == nil {
-				statusLines := strings.Split(status, "\n")
-				status = strings.Join(statusLines[1:], "\n")
-				err := json.Unmarshal([]byte(status), &statusOutputJSON)
-				require.NoError(tt, err)
-				if runtimeStatus, ok := statusOutputJSON["runtimeSecurityStatus"]; ok {
-					if connected, ok := runtimeStatus.(map[string]any)["connected"]; ok {
-						result = connected == true
-					}
-				}
+			if err != nil {
+				tt.Logf("[CheckCWSBehaviour] Try #%d: unable to retrieve security-agent status with error %s", try, err)
+				time.Sleep(1 * time.Second)
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
You could use `testify.Eventually` instead of a custom retry https://pkg.go.dev/github.com/stretchr/testify/assert#EventuallyWithT